code/app/global_variables.py

```diff
@@ -41,31 +134,155 @@
 
 
 for idx in np.arange(len(LGL_list)):
-	LGL_list[idx] = af.arith.cast(af.Array(LGL_list[idx]), af.Dtype.f64)
-	pass
+	LGL_list[idx] = np.array(LGL_list[idx], dtype = np.float64)
+	LGL_list[idx] = af.interop.np_to_af_array(LGL_list[idx])
+
+for idx in np.arange(len(gaussianNodesList)):
+	gaussianNodesList[idx] = np.array(gaussianNodesList[idx], \
+										dtype = np.float64)
+	gaussianNodesList[idx] = af.interop.np_to_af_array(gaussianNodesList[idx])
 
-N_LGL       = 16
-xi_LGL      = None
-lBasisArray = None
+x_nodes         = af.interop.np_to_af_array(np.array([-1., 1.]))
+N_LGL           = 16
+xi_LGL          = None
+lBasisArray     = None
+lobatto_weights = None
+N_Elements      = None
+element_nodes   = None
+u               = None
+time            = None
+c               = None
 
-def populateGlobalVariables(N = 16):
+
+def populateGlobalVariables(Number_of_LGL_pts = 8, Number_of_elements = 10):
 	'''
-	Initialize the global variables
-	
+	Initialize the global variables.
 	Parameters
 	----------
-	N : int
-		Number of LGL points.
+	N_LGL : int
+			Number of LGL points.
+			Declares the number and the value of
+	
+	N_gauss : int
+			  Number of gaussian nodes required.
+	
 	'''
 	
 	global N_LGL
 	global xi_LGL
 	global lBasisArray
-	
-	N_LGL       = N
+	global lobatto_weights
+	N_LGL       = Number_of_LGL_pts
 	xi_LGL      = lagrange.LGL_points(N_LGL)
-	
 	lBasisArray = af.interop.np_to_af_array( \
 		lagrange.lagrange_basis_coeffs(xi_LGL))
 	
+	lobatto_weights = af.interop.np_to_af_array(\
+		lobatto_weight_function(N_LGL, xi_LGL)) 
+	
+	global N_Elements
+	global element_nodes
+	N_Elements       = Number_of_elements
+	element_size     = af.sum((x_nodes[1] - x_nodes[0]) / N_Elements)
+	elements_xi_LGL  = af.constant(0, N_Elements, N_LGL)
+	elements         = utils.linspace(af.sum(x_nodes[0]), \
+		af.sum(x_nodes[1] - element_size), N_Elements)
+	
+	np_element_array = np.concatenate((af.transpose(elements), 
+						   af.transpose(elements + element_size)))
+	
+	element_array = (af.transpose(af.interop.np_to_af_array(np_element_array)))
+	element_nodes = af.transpose(wave_equation.mappingXiToX(\
+										af.transpose(element_array), xi_LGL))
+	
+	
+	global u
+	global time
+	u_init     = np.e ** (-(element_nodes) ** 2 / 0.4 ** 2)
+	time       = utils.linspace(0, 2, 20)
+	u          = af.constant(0, (N_Elements), N_LGL, time.shape[0])
+	u[:, :, 0] = u_init
+	
+	global c
+	c = 1.0
+	
 	return
+
+
+def gaussian_weights(N, i):
+	'''
+	Returns the gaussian weights for :math: `N` Gaussian Nodes at index
```

Comment:
Give the formula for calculating the Gaussian weights.
Give a wiki reference for that formula.